experiment/secret-sync-controller/secret-sync-config.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+// configuration and sync-pair defination
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"gopkg.in/yaml.v2"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Structs for configuration
+type SecretSyncConfig struct {
+	Specs []SecretSyncSpec `yaml:"specs"`
+}
+
+type SecretSyncSpec struct {
+	// the Source is always a SecretManager secret
+	// the Destination is always a Kubernetes secret
+	Source      SecretManagerSpec `yaml:"source"`
+	Destination KubernetesSpec    `yaml:"destination"`
+}
+
+type KubernetesSpec struct {
+	Namespace string `yaml:"namespace"`
+	Secret    string `yaml:"secret"`
+	Key       string `yaml:"key"`
+}
+
+type SecretManagerSpec struct {
+	Project string `yaml:"project"`
+	Secret  string `yaml:"secret"`
+}
+
+func (config SecretSyncConfig) String() string {
+	d, _ := yaml.Marshal(config)
+	return string(d)
+}
+
+type SecretData struct {
+	Data []byte
+	// Potentially more information
+}
+
+// structs for client interface
+type ClientInterface interface {
+	GetKubernetesSecret(KubernetesSpec) (*SecretData, error)
+	GetSecretManagerSecret(SecretManagerSpec) (*SecretData, error)
+	UpsertKubernetesSecret(KubernetesSpec, *SecretData) (*SecretData, error)
```

Comment:
I was trying to avoid copying of potentially large SecretData.
But I think I'll just use []byte.
It would be better to put all these in the `client` package.